p2p/net/swarm/swarm_listen.go

```diff
@@ -94,8 +122,15 @@ func (s *Swarm) AddListenAddr(a ma.Multiaddr) error {
 			c, err := list.Accept()
 			if err != nil {
 				if s.ctx.Err() == nil {
-					// only log if the swarm is still running.
-					log.Errorf("swarm listener accept error: %s", err)
+					s.listeners.Lock()
+					_, ok := s.listeners.m[list]
+					s.listeners.Unlock()
+
+					if ok {
+						// only log if the swarm is still running and the
+						// listener has not been closed intentionally.
+						log.Errorf("swarm listener accept error: %s", err)
+					}
```

Comment:
1. We don't need to check `s.ctx` anymore.
2. We can move this into the defer. If the listener ever closes without being removed from the map, that means it wasn't intentionally closed and we should log.

Reply:
Done, I also changed the log to a more explicit message: `swarm listener unintentionally closed`.